### DISCUSSION TITLE
Extra lines in osgFX::Outline

### DISCUSSION THREAD

**Discussion by korimac** [openscenegraph/OpenSceneGraph#1316] [Q&A] [answered]
Hello,

I'm currently using osgX::Outline to show which model is currently selected by a user. When testing this on various files, there appear to be extraneous lines connecting some of the vertices. I included a sample image with an unmodified Stanford Bunny OBJ below. Is there anything I can do to remove these lines and produce a clean outline? 

![image](https://github.com/openscenegraph/OpenSceneGraph/assets/37956812/2c1e7eac-584f-4a34-a475-c570b8f5d274)

Any nudge in the right direction is appreciated. Thank you very much,
Kori

**Comment by robertosfield** [comment, marked as answer]
At a guess there could be an issue in the mesh that it highlighted by forcing to wireframe for the outline.  vsgFX::Outline is very basic technique, written almost 2 decades ago, it might not be the best tool for the job these days.

**Comment by korimac** [comment]
Thank you for the quick reply, Robert! 

That makes sense. Is there a preferred technique to generate an outline that is more recent within OSG? Unfortunately I am unable to upgrade to VulkanSceneGraph at this time as this is a smaller component of a larger OSG project.

**Comment by robertosfield** [comment]
How to implement it will depend on the rest of your application/data.  I know nothing about that so can't comment in detail.  Go look online for different techniques for rendering outlines and see what fits your application/data usage case best.

**Comment by AnyOldName3** [comment]
I read https://bgolus.medium.com/the-quest-for-very-wide-outlines-ba82ed442cd9 recently, which has interesting information.

**Comment by korimac** [comment]
Thank you for sharing! It turns out this was a bit more of a complex problem than I was originally envisioning, but I'm seeing a few ways it can get done now. Appreciate it!
